Source/WebCore/workers/WorkerMessagingProxy.cpp

```diff
@@ -325,21 +330,17 @@ void WorkerMessagingProxy::setResourceCachingDisabledByWebInspector(bool disable
 
 void WorkerMessagingProxy::workerThreadCreated(DedicatedWorkerThread& workerThread)
 {
+    ASSERT(!m_askedToTerminate);
     m_workerThread = &workerThread;
 
-    if (m_askedToTerminate) {
-        // Worker.terminate() could be called from JS before the thread was created.
-        m_workerThread->stop(nullptr);
-    } else {
-        if (m_askedToSuspend) {
-            m_askedToSuspend = false;
-            m_workerThread->suspend();
-        }
-
-        auto queuedEarlyTasks = WTFMove(m_queuedEarlyTasks);
-        for (auto& task : queuedEarlyTasks)
-            m_workerThread->runLoop().postTask(WTFMove(*task));
+    if (m_askedToSuspend) {
+        m_askedToSuspend = false;
+        m_workerThread->suspend();
     }
+
+    auto queuedEarlyTasks = WTFMove(m_queuedEarlyTasks);
```

Comment:
Because we are touching this it should use be changed to `std::exchange(m_queuedEarlyTasks, { })` rather than `WTFMove` since it’s relying on `m_queuedEarlyTasks` becoming empty.

Reply:
Done